src/st/no3/48A84.c

```diff
@@ -246,7 +246,135 @@ INCLUDE_ASM("st/no3/nonmatchings/48A84", EntityStrongWargWaveAttack);
 INCLUDE_ASM("st/no3/nonmatchings/48A84", EntityUnkId2F);
 
 // beams that go up when strong warg dies
-INCLUDE_ASM("st/no3/nonmatchings/48A84", EntityStrongWargDeathBeams);
+void EntityStrongWargDeathBeams(Entity* self) {
+    Primitive* prim;
+    s16 baseX;
+    u16 hiddenPrimCount;
+    u16 palette;
+    s32 primIndex;
+    s32 priorityOffset;
+    s32 temp_s1;
+    u16 temp_s1_u16;
+
+    switch (self->step) {
+    case 0:
+        palette = self->palette + 4;
+        temp_s1 = self->unk5A + 3;
+        temp_s1_u16 = (u16)temp_s1;
+
+        InitializeEntity(g_eInitGeneric2);
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 4);
+
+        if (primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->primIndex = primIndex;
+        self->flags |= FLAG_HAS_PRIMS;
+        prim = &g_PrimBuf[primIndex];
+
+        while (prim != NULL) {
+            prim->tpage = temp_s1_u16 / 4;
+            prim->clut = palette;
+            prim->u0 = prim->u1 = (temp_s1 & 1) << 7 | 0x21;
+            prim->v1 = prim->v3 = (temp_s1 & 2) << 6 | 0x59;
+            prim->v0 = prim->v2 = (temp_s1 & 2) << 6 | 0x7F;
```

Comment:
Do these still match if you use + instead of | at the end? Since UV coordinates are numerical values, it probably makes more sense to add a physical offset than to bitwise OR it.